pennylane/ops/qubit/matrix_ops.py

```diff
@@ -290,9 +306,42 @@ def compute_decomposition(U: TensorLike, wires: WiresLike):
                 raise DecompositionUndefinedError(
                     "The decomposition of a two-qubit sparse QubitUnitary is undefined."
                 )
+
             return qml.ops.two_qubit_decomposition(U, Wires(wires))
 
-        return super(QubitUnitary, QubitUnitary).compute_decomposition(U, wires=wires)
+        ops = []
+
+        # Combining the two equalities in Fig 14 [https://arxiv.org/pdf/quant-ph/0504100], we can express
+        # a n-qubit unitary U with four (n-1)-qubit unitaries and three multiplexers (qml.SelectPauliRot)
+
+        (u1, u2), theta, (v1_dagg, v2_dagg) = sp.linalg.cossin(
```

Comment:
Could we make this jax-jit compatible with `pure_callback`? Not sure if it is the best way.